lighthouse-core/report/html/renderer/util.js

```diff
@@ -72,6 +72,21 @@ class Util {
       }
     }
 
+    if (clone.stackPacks) {
+      for (const audit of Object.values(clone.audits)) {
+        audit.stackPacks = {};
+        clone.stackPacks.forEach(pack => {
+          // if advice exist for this audit
+          if (pack.advice[audit.id]) {
+            audit.stackPacks[pack.id] = {
```

Comment:
by adding stackpack to the audit object we don't have to change any code in the report. Downside is that we duplicate the icon for each audit this could be a memory issue?

If it's a problem we might want to put stackpacks on global scope, create a context object that we can pass down the tree or just pass stackpacks down the tree until it get rendered.

Reply:
> Downside is that we duplicate the icon for each audit this could be a memory issue?

The reference to the string is duplicated but the string itself shouldn't be multiplying in memory, so I think we're fine here.


Another question is if we want to continue mutating the audit object to pass information down the chain. I sympathize with the desire to do this because threading down additional context everywhere is a pain, but at some point it's going to be worth splitting our `UIAudit` and `AudiResult` types

Reply:
> The reference to the string is duplicated but the string itself shouldn't be multiplying in memory, so I think we're fine here.

wasn't 100% sure how this was handled by v8, I thought primitive types as string would copy their values instead of doing a reference 🤷‍♂️